### DISCUSSION TITLE
Images from markdown documentation, are imported into workspace.json, but never removed.

### DISCUSSION THREAD

**Discussion by gumbo2k** [structurizr/lite#161] [Q&A]
### Expected

When linking to markdown documentation of a software system with images, the documentation is rendered correctly in structurizr.
A side effect is, that the image gets copied and embedded as base64 into the `workspace.json` file.

**Expectation:** When removing the image-link from the documentation, I would expect structurizr to also remove the embedded image from the workspace.json.

**Alternative Expectation:** Not embedd the image into workspace.json at all. I suspect, it is needed "on the wire" between the structurizr server and the browser, but at least not the workspace.json that is stored on disk could be cleaned up.

### Actual

The image stays embedded as base64 string in the workspace.json

### Steps to reproduce

Create a `workspace.dsl` and link the software system documentation
```
...
        mysystem = softwareSystem "My Software System" {
            !docs ../../arc42/mysystem
...
```
Create a the documentation file `../../arc42/mysystem/architecture.md` that contains an image link like this:
```
...
![Some Image](images/some-image.png)
...
```
Place `some-image.png` in ../../arc42/mysystem/images/

Open workspace with structurizr lite (if you use the container, make sure you do not bind-mount the directory where `workspace.dsl` lies, but its grand-parent directory and use `STRUCTURIZR_WORKSPACE_PATH=dir1/dir2` )

### Version/build information

 Structurizr Lite
  - build: 2024.07.02 (2024-07-02T12:09:42Z)
  - structurizr-java: v2.2.0
  - structurizr-dsl: v2.2.0
  - structurizr-import: v2.2.0
  - structurizr-graphviz: v2.2.0

### Severity

Minor

### Priority

I have no budget and there's no rush, please fix this for free

**Comment by simonbrowndotje** [comment]
This is working as expected ... file-based images are inlined (as a Base64 data URIs) into the workspace. The reason is that Markdown is rendered client-side in the web browser, which doesn't have access to your file system.

**Reply by gumbo2k** [reply]
Hello @simonbrowndotje , I understand the technical necessity of including the image in the workspace.json. :+1: 

The bug I reported was about _removing the images when they are not referenced anymore_.

However, I just found that structurizr **will** remove embedded images from workspace.json . It just does not do it based on them being *referenced in the documentation* , but on the files being *present on the server side filesystem* in the `!docs` directory. :grin: 


So there is no actual bug. However there is potential for removing some redundancy.

AFAIK, the documentation (including the inlined images) and a base64 encoded version of `workspace.dsl` file are never edited by the client, thus they could be removed from `workspace.json` after it has been sent to the client and before it is persisted on the server side, correct?

**Reply by simonbrowndotje** [reply]
> However there is potential for removing some redundancy.

Correct - all images underneath the documentation folder will be imported into the workspace. So if removing images from the folder isn't possible, I'd probably recommend writing a [DSL plugin](https://docs.structurizr.com/dsl/plugins) that removes any unused images.
